### PR TITLE
feat: add getters to builders

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**

This PR adds back getters to the builder classes in discord.js that were removed in #7584 as that PR didn't take into account the possibility of creating agnostic functions that accept builder instances as input and modify them according to their props. 
I also fixed the docs in some places related to this (aka component classes)

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by monbrey** [comment]
Why can't the agnostic functions access `<Builder>.data.property`?

**Comment by ImRodry** [comment]
Because those properties are snake_cased

**Comment by monbrey** [comment]
> Because those properties are snake_cased

I might be missing something but that doesn't seem to address why those properties couldn't be used, unless you consider that to no longer be "agnostic"?

**Comment by ImRodry** [comment]
There are no properties in discord.js that are snake_cased, so these shouldn't be any different, it's purely for the sake of consistency

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -541,35 +544,62 @@ export class ButtonComponent extends Component<APIButtonComponent> {
...
 export class TextInputBuilder extends BuilderTextInputComponent {
   public constructor(data?: TextInputComponentData | APITextInputComponent);
   public static from(other: JSONEncodable<APITextInputComponent> | APITextInputComponent): TextInputBuilder;
+  public get label(): string | null;
+  public get style(): TextInputStyle | null;
+  public get maxLength(): number | null;
+  public get minLength(): number | null;
+  public get required(): boolean | null;
+  public get value(): string | null;
+  public get placeholder(): string | null;
```
Comment (vitorlops) [inline packages/discord.js/typings/index.d.ts:595]: It's missing `customId`.

**Comment by ImRodry** [comment]
That was a very personal comment but thank you for that. I understand your points and I believe my code could be adjusted to do what you suggested and it would even look better. I will close this PR now as I understand it is not needed and will open a separate one simply to document the missed typedefs and classes.

**Comment by ImRodry** [comment]
Turns out the comment above addressed one situation while still leaving room for others where this PR is needed. I'm reopening this because other people might face similar issues as I have and this will facilitate their task to upgrade a lot.

**Comment by suneettipirneni** [comment]
> Turns out the comment above addressed one situation while still leaving room for others where this PR is needed.

Which you already were given more-than valid solution(s) too.

**Comment by ImRodry** [comment]
If you could read the full comment I’d appreciate it

**Comment by suneettipirneni** [comment]
Second half of the comment is a moot point considering its basis (the first half) is misleading.

**Comment by ImRodry** [comment]
I'm making this PR to improve the library, not for myself

**Comment by ImRodry** [comment]
@iCrawl would appreciate an explanation on why this was closed seeing as it even had positive reactions

**Comment by iCrawl** [comment]
Because we have no plans to re-introduce them and here's plenty of explanation above.

**Pull request closed without merging**